sycl/include/sycl/info/aspects.def

```diff
@@ -12,27 +11,27 @@ __SYCL_ASPECT(queue_profiling, 12)
 __SYCL_ASPECT(usm_device_allocations, 13)
 __SYCL_ASPECT(usm_host_allocations, 14)
 __SYCL_ASPECT(usm_shared_allocations, 15)
-__SYCL_ASPECT(usm_restricted_shared_allocations, 16)
-__SYCL_ASPECT(usm_system_allocations, 17)
-__SYCL_ASPECT(ext_intel_pci_address, 18)
-__SYCL_ASPECT(ext_intel_gpu_eu_count, 19)
-__SYCL_ASPECT(ext_intel_gpu_eu_simd_width, 20)
-__SYCL_ASPECT(ext_intel_gpu_slices, 21)
-__SYCL_ASPECT(ext_intel_gpu_subslices_per_slice, 22)
-__SYCL_ASPECT(ext_intel_gpu_eu_count_per_subslice, 23)
-__SYCL_ASPECT(ext_intel_max_mem_bandwidth, 24)
-__SYCL_ASPECT(ext_intel_mem_channel, 25)
-__SYCL_ASPECT(usm_atomic_host_allocations, 26)
-__SYCL_ASPECT(usm_atomic_shared_allocations, 27)
-__SYCL_ASPECT(atomic64, 28)
-__SYCL_ASPECT(ext_intel_device_info_uuid, 29)
-__SYCL_ASPECT(ext_oneapi_srgb, 30)
-__SYCL_ASPECT(ext_oneapi_native_assert, 31)
-__SYCL_ASPECT(host_debuggable, 32)
-__SYCL_ASPECT(ext_intel_gpu_hw_threads_per_eu, 33)
-__SYCL_ASPECT(ext_oneapi_cuda_async_barrier, 34)
-__SYCL_ASPECT(ext_oneapi_bfloat16_math_functions, 35)
-__SYCL_ASPECT(ext_intel_free_memory, 36)
-__SYCL_ASPECT(ext_intel_device_id, 37)
-__SYCL_ASPECT(ext_intel_memory_clock_rate, 38)
-__SYCL_ASPECT(ext_intel_memory_bus_width, 39)
+__SYCL_ASPECT(usm_system_allocations, 16)
```

Comment:
I'm concerned a bit about changing values of aspects here, it could lead to ABI breaks, see https://github.com/intel/llvm/pull/3013. 

I would recommend to just remove an enumerator, leaving other values as is.

Also, there is a slight concern about removing and aspect without deprecating it first. However, I'm not entirely sure if deprecating worth it, because this aspect has never been in SYCL 2020 in the first place, if I understand correctly.